agent/reconciler/host_reconciler.go

```diff
@@ -289,10 +290,11 @@ func (r *HostReconciler) installK8sComponents(ctx context.Context, byoHost *infr
 	bundleRegistry := byoHost.GetAnnotations()[infrastructurev1beta1.BundleLookupBaseRegistryAnnotation]
 	k8sVersion := byoHost.GetAnnotations()[infrastructurev1beta1.K8sVersionAnnotation]
 	byohBundleTag := byoHost.GetAnnotations()[infrastructurev1beta1.BundleLookupTagAnnotation]
-	bundleInstaller, err := installer.New(bundleRegistry, r.DownloadPath, logger)
+	bundleInstaller, err := installer.New(r.DownloadPath, logger)
 	if err != nil {
 		return err
 	}
+	bundleInstaller.SetBundleRepo(bundleRegistry)
```

Comment:
Can we move this `bundleRegistry` to be part of the `Install` contract?

Why? - Because in the current implementation, we are relying on the caller to remember to call `SetBundleRepo` before calling the `Install` method. Lets just make it part of the `Install` method, so that the caller knows that this is the information needed before installing.